modules/eks/cluster/variables.tf

```diff
@@ -398,3 +398,12 @@ variable "addons" {
   description = "Manages [`aws_eks_addon`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_addon) resources"
   default     = []
 }
+
+variable "addons_depends_on" {
+  type        = bool
+  description = <<-EOT
+    If provided, all addons will depend on node groups and therefore not be installed until nodes are finalized.
```

Comment:
```suggestion
    If set `true`, all addons will depend on managed node groups provisioned by this component and therefore not be installed until nodes are provisioned.
```

Reply:
Added a warning that addons may not work with Karpenter to the section in the readme